File: job_portal/src/app/core/models/job.model.ts
export interface Job {
  id: string; // UUID
  title: string;
  company?: string; // For backward compatibility
  company_name?: string; // Database field
  rating?: number; // For backward compatibility
  company_rating?: number; // Database field
  reviews?: number | null; // For backward compatibility
  company_reviews_count?: number; // Database field
  location?: string;
  experience?: string; // For backward compatibility
  experience_level?: string; // Database field
  employment_type?: string; // Database field (Full-time, Part-time, etc.)
  category?: string; // Database field
  status?: string; // Database field (active, inactive, etc.)
  salary?: number; // For backward compatibility
  salary_min?: number; // Database field in LPA
  salary_max?: number; // Database field in LPA
  salary_currency?: string; // Database field
  salaryRange?: string; // For display
  postedDate?: string; // For sorting
  created_at?: string; // Database field
  updated_at?: string; // Database field
  expires_at?: string | null; // Database field
  summary?: string;
  description?: string | any; // Can be string (database) or object (frontend)
  requirements?: string; // Database field
  companyType?: string; // For backward compatibility
  company_type?: string; // Database field
  company_size?: string; // Database field
  tags?: string[]; // For backward compatibility
  skills_required?: string; // Database field
  benefits?: string; // Database field
  posted?: string; // For backward compatibility
  posted_by?: string; // Database field - user ID who posted
  views?: number; // Database field
  remote_allowed?: boolean; // Database field
  logo?: string | null; // For backward compatibility
  logoText?: string; // For backward compatibility
  color?: string; // For backward compatibility
  // Legacy description object structure for backward compatibility
  meta?: {
    role?: string;
    industry?: string;
    department?: string;
    employment?: string;
    education?: {
      UG?: string;
      PG?: string;
    };
  };
  responsibilities?: string[];
  qualifications?: string[];
  skills?: string[];
}

File: job_portal_fastapi/app/db/models/job/job_model.py
from sqlalchemy import Column, String, Text, Integer, DateTime, Numeric, Boolean, ForeignKey
from sqlalchemy.orm import relationship
from app.db.base import BaseModel

class Job(BaseModel):
    __tablename__ = 'jobs'
    title = Column(String(255))
    company_name = Column(String(255))
    location = Column(String(255))
    employment_type = Column(String(100))
    experience_level = Column(String(100))
    category = Column(String(100))
    status = Column(String(50), default='active')
    description = Column(Text)
    requirements = Column(Text)
    posted_by = Column(String(36), ForeignKey('users.id'))
    views = Column(Integer, default=0)
    expires_at = Column(DateTime)
    salary_min = Column(Numeric(10, 2))
    salary_max = Column(Numeric(10, 2))
    salary_currency = Column(String(3), default='INR')
    company_rating = Column(Numeric(2, 1))
    company_reviews_count = Column(Integer, default=0)
    remote_allowed = Column(Boolean, default=False)
    skills_required = Column(Text)
    benefits = Column(Text)
    company_size = Column(String(50))
    company_type = Column(String(50), default='Corporate')

    poster = relationship("User")
